### DISCUSSION TITLE
Possible bug reading JP2 as grayscale when should be color

### DISCUSSION THREAD

**Discussion by fmw42** [ImageMagick/ImageMagick#6671] [Contribute]
### ImageMagick version

7.1.1.6

### Operating system

MacOS

### Operating system, version and so on

Monterey

### Description

The attached image seems to be read as grayscale with alpha in Imagemagick when I believe it should be in color.

identify -verbose says:

```
 Channel depth:
    gray: 8-bit
    alpha: 8-bit

```

But Exiftool says:

```
 0 ImageHeight                     : 10000
    4 ImageWidth                      : 10000
    8 NumberOfComponents              : 4

but it also says

    3 ColorSpace                      : Grayscale

```

NumberOfComponents I interpret as having 4 channels, RGBA, but then it says colorspace is grayscale.  So I am confused about this file. 

### Steps to Reproduce

See above

### Images

http://www.fmwconcepts.com/misc_tests/dop10rgbi_32_280_5652_1_nw_2021.jp2.zip

Original from first file at https://www.opengeodata.nrw.de/produkte/geobasis/lusat/dop/dop_jp2_f10/

**Comment by snibgo** [comment]
JPEG 2000 files don't use the RGB color model. See https://en.wikipedia.org/wiki/JPEG_2000

I don't know why exiftool says there are 4 components.

Gimp shows the image as grayscale.

**Comment by fmw42** [comment]
Oddly, then, Imagemagick identify says

```
magick identify dop10rgbi_32_280_5652_1_nw_2021.jp2
dop10rgbi_32_280_5652_1_nw_2021.jp2 JP2 10000x10000 10000x10000+0+0 8-bit sRGB 0.000u 0:00.000
```

So why sRGB and not Grayscale?

**Comment by fmw42** [comment]
GraphicConverter opens it as color with alpha. But the colors are washed out.

**Comment by fmw42** [comment]
Photoshop opens it as grayscale.  Python Pillow opens it as proper color.

If the file actually has 4 channels, the 4th may be an IR channel rather than an alpha.

**Comment by fmw42** [comment]
The meta data indicate that the 4th channel is indeed a near IR channel.  

Since there are 4 real channels (not an alpha), this would be a good image for reading into 4 channels and not an alpha. It would also be good for PCA analysis to reduce to three PCA 3 channels with the most information. Also for displaying different combinations of 3 of the 4 channels with -channel-fx or with -ftxt.

**Comment by fmw42** [comment]
Any news about verifying the color vs grayscale reading issue? Can anyone else verify this issue? It should be read in as color and not grayscale for the colorspace.

**Comment by fmw42** [comment]
**I still think this is still a bug.**

The image is color with an IR channel. So properly it would be color with a meta channel. But as IM does not know it is meta, it should be read as color with alpha.  

It should not be read as grayscale.
